tomni/annotation_manager/test_annotation_manager.py

```diff
@@ -674,107 +680,542 @@ def test_binary_mask_multiple_objects_not_connected_to_labeled_mask_connectivity
+                "accuracy": 1,
+            },
+        ]
+
+        label = "alive"
+        manager = AnnotationManager.from_labeled_mask(
+            data, include_inner_contours=True, labels=label
+        )
+        actual = manager.to_dict(features=[])
+        for dict_object in actual:
+            dict_object.pop("id")
+        np.testing.assert_array_equal(actual, expected)
+
+    def test_single_label_exclude_inner(self):
+        data = np.array(
+            [
+                [0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1],
+                [0, 1, 0, 0, 2, 2, 2, 2, 2, 2, 0, 0, 1, 0, 0],
+                [0, 1, 0, 0, 2, 2, 2, 2, 2, 2, 0, 0, 1, 0, 0],
+                [0, 1, 0, 0, 2, 2, 2, 2, 2, 2, 0, 0, 1, 0, 0],
+                [0, 1, 0, 0, 0, 2, 2, 2, 2, 0, 0, 0, 1, 0, 0],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
+                [0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+            ]
+        )
+
+        expected = [
+            {
+                "label": "alive",
+                "children": [],
+                "parents": [],
+                "type": "polygon",
+                "points": [
+                    {"x": 1, "y": 0},
+                    {"x": 1, "y": 8},
+                    {"x": 12, "y": 8},
+                    {"x": 12, "y": 3},
+                    {"x": 13, "y": 2},
+                    {"x": 14, "y": 2},
+                    {"x": 14, "y": 0},
+                ],
+                "inner_points": [],
+                "accuracy": 1,
+            },
+        ]
+
+        label = "alive"
+        manager = AnnotationManager.from_labeled_mask(
+            data, include_inner_contours=False, labels=label
+        )
+        actual = manager.to_dict(features=[])
+        for dict_object in actual:
+            dict_object.pop("id")
+        np.testing.assert_array_equal(actual, expected)
+
+    def test_labeled_mask_classes_mismatch(self):
+        data = np.array(
+            [
+                [0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1],
+                [0, 1, 0, 0, 2, 2, 2, 2, 2, 2, 0, 0, 1, 0, 0],
+                [0, 1, 0, 0, 2, 2, 2, 2, 2, 2, 0, 0, 1, 0, 0],
+                [0, 1, 0, 0, 2, 2, 2, 2, 2, 2, 0, 0, 1, 0, 0],
+                [0, 1, 0, 0, 0, 2, 2, 2, 2, 0, 0, 0, 1, 0, 0],
+                [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
+                [0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+            ]
+        )
+
+        labels = ["alive"]
+
+        self.assertRaises(
+            AssertionError,
+            AnnotationManager.from_labeled_mask,
+            data,
+            labels=labels,
+            include_inner_contours=True,
+        )
+
+    def test_labeled_mask_classes_mismatch_2(self):
```

Comment:
this should not fail. if model expects to predict 2 labels but on 1 image it only sees one label it should not fial